apps/aechannel/test/aesc_fsm_SUITE.erl

```diff
@@ -2351,22 +2460,34 @@ create_channel_from_spec(I, R, Spec, Port, UseAny, Debug, Cfg) ->
 
     I3 = await_funding_locked(I2, ?TIMEOUT, Debug),
     R3 = await_funding_locked(R2, ?TIMEOUT, Debug),
-    log(Debug, "=== Funding locked ===", []),
+    ?LOG(Debug, "=== Funding locked ===", []),
     %%
     I4 = await_update_report(I3, ?TIMEOUT, Debug),
     R4 = await_update_report(R3, ?TIMEOUT, Debug),
-    log(Debug, "=== Update reports received ===", []),
+    ?LOG(Debug, "=== Update reports received ===", []),
     I5 = await_open_report(I4, ?TIMEOUT, Debug),
     R5 = await_open_report(R4, ?TIMEOUT, Debug),
-    log(Debug, "=== Open reports received ===", []),
-    log(Debug, "=== Message Q: ~p", [element(2,process_info(self(), messages))]),
+    ?LOG(Debug, "=== Open reports received ===", []),
+    ?PEEK_MSGQ,
     [] = check_info(0, Debug),
-    #{i => I5, r => R5, spec => Spec}.
+    trace_checkpoint(?TR_CHANNEL_CREATED, Cfg),
+    maps:merge(#{i => I5, r => R5, spec => Spec},
+               maps:from_list([{K,V} || {K,V} <- Cfg,
+                                        lists:member(K, [initiator_opts,
+                                                         responder_opts])])).
+
+customize_spec(Role, Spec, Cfg) ->
+    maps:merge(Spec, custom_spec_opts(Role, Cfg)).
+
+custom_spec_opts(responder, Cfg) ->
+    proplists:get_value(responder_opts, Cfg, #{});
+custom_spec_opts(initiator, Cfg) ->
+    proplists:get_value(initiator_opts, Cfg, #{}).
 
 spawn_responder(Port, Spec, R, UseAny, Debug) ->
     Me = self(),
-    spawn(fun() ->
-                       log("responder spawned: ~p", [Spec]),
+    spawn_link(fun() ->
```

Comment:
In case a rare CI crash occurs in the child process the crash will kill the test process instead of retrying the operation :)